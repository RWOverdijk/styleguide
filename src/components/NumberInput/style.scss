@import "../../css/_global.scss";
@import "../../css/_form-globals.scss";

$number-input-ctrl-width:       50px;
$number-input-color:            $color-pink-primary;
$number-input-ctrl-hover-color: $color-pink-primary;
$number-input-ctrl-bg:          $color-grey-light;

.Van-NumberInput {
  border: 1px solid transparent;
  border-radius: 5px;
  overflow: hidden;
  position: relative;
  width: 160px;
  z-index: 2;

  &--error {
    border: 1px solid $input-error-border;
  }

  .Van-NumberInput-input {
    -moz-appearance: textfield;
    -webkit-appearance: textfield;
    border: 0;
    box-sizing: border-box;
    color: $color-pink-primary;
    appearance: textfield;
    background: $input-bg;
    cursor: text;
    display: block;
    font-size: $input-font-size;
    padding: $input-padding-top ($number-input-ctrl-width + $input-padding-left);
    text-align: center;
    width: 100%;

    // fix number input controls in chrome
    &::-webkit-inner-spin-button,
    &::-webkit-outer-spin-button {
      -webkit-appearance: textfield;
      -moz-appearance: textfield;
      appearance: textfield;
    }

    &:focus {
      outline: 0;
    }
  }

  .Van-NumberInput-ctrl {
    background-color: $number-input-ctrl-bg;
    border: 0;
    bottom: 0;
    color: #5b636b;
    cursor: pointer;
    font-size: 23px;
    font-weight: 900;
    position: absolute;
    text-align: center;
    top: 0;
    width: $number-input-ctrl-width;

    &:focus {
      outline: 0;
    }

    &:active {
      outline: 0;
      color: $number-input-ctrl-hover-color;
    }
  }

  .Van-NumberInput-ctrl--up {
    border-radius: 0 5px 5px 0;
    line-height: 1.4;
    right: 0;
  }

  .Van-NumberInput-ctrl--down {
    border-radius: 5px 0 0 5px;
    left: 0;
  }
}
