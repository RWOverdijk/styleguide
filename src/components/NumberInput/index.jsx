import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './style.scss';

class NumberInput extends React.Component {
  static defaultProps = {
    input: {},
    meta: {},
    step: 1
  };

  static propTypes = {
    input: PropTypes.object,
    meta: PropTypes.object,
    step: PropTypes.number
  };

  handleClick = (dir) => {
    const { step = 1 } = this.props;
    const val = this.input.value.length ? parseInt(this.input.value, 10) : 0;

    this.input.value = dir === 'up'
      ? val + step
      : val - step;
  }

  render() {
    const {
      input,
      meta: { error, touched },
      ...props
    } = this.props;
    const className = classnames(
      'Van-NumberInput',
      { 'Van-NumberInput--error': touched && error }
    );

    return (
      <div className={className}>
        <button type="button" className="Van-NumberInput-ctrl Van-NumberInput-ctrl--down" onClick={() => this.handleClick('down')}>-</button>
        <button type="button" className="Van-NumberInput-ctrl Van-NumberInput-ctrl--up" onClick={() => this.handleClick('up')}>+</button>

        <input
          {...props}
          {...input}
          ref={(numinput) => { this.input = numinput; }}
          type="number"
          className="Van-NumberInput-input"
        />
      </div>
    );
  }
}

export default NumberInput;
