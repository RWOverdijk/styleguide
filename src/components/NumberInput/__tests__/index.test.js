import React from 'react';
import renderer from 'react-test-renderer';
import NumberInput from '../';

describe('<NumberInput/>', () => {
  it("renders the correct markup", () => {
    const tree = renderer.create(
      <NumberInput />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <NumberInput className="my-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <NumberInput data-id="0" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
