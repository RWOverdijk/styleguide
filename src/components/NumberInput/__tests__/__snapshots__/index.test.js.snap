// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberInput/> renders the correct markup 1`] = `
<div
  className="Van-NumberInput"
>
  <button
    className="Van-NumberInput-ctrl Van-NumberInput-ctrl--down"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="Van-NumberInput-ctrl Van-NumberInput-ctrl--up"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <input
    className="Van-NumberInput-input"
    step={1}
    type="number"
  />
</div>
`;

exports[`<NumberInput/> renders with specific attributes 1`] = `
<div
  className="Van-NumberInput"
>
  <button
    className="Van-NumberInput-ctrl Van-NumberInput-ctrl--down"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="Van-NumberInput-ctrl Van-NumberInput-ctrl--up"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <input
    className="Van-NumberInput-input"
    data-id="0"
    step={1}
    type="number"
  />
</div>
`;

exports[`<NumberInput/> renders with specific className 1`] = `
<div
  className="Van-NumberInput"
>
  <button
    className="Van-NumberInput-ctrl Van-NumberInput-ctrl--down"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="Van-NumberInput-ctrl Van-NumberInput-ctrl--up"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <input
    className="Van-NumberInput-input"
    step={1}
    type="number"
  />
</div>
`;
